ATM.py

```diff
@@ -44,97 +41,81 @@ def check_password(user_id, input_password, pass_data):
+
+    def update_data(self):
+        with open(BANK_DATA_PATH, 'w') as new_bank:
+            for key in self.pass_dict:
+                new_bank.write(f"{key} {self.pass_dict[key]} "
+                               f"{self.balance_dict[key]}\n")
 
 
 def main():
     user = ATM()
-
-    pass_dict = user.get_passwords_dictionary()
-    bal_dict = user.get_balance_dictionary()
     print('hello and welcome to The Thieves ATM!')
     id_num = input('please enter you ID number\n')
-    if user.check_id(id_num):
-        password = input('please enter your password\n')
-        user.check_password(id_num, password, pass_dict)
-        print('What would you like to do today?\n'
-              'To check your balance press 1\n'
-              'To deposit money press 2\n'
-              'To withdraw money press 3\n'
-              'To change password press 4\n'
-              'for Exit press -1\n')
-        choose = input('\n')
-        while choose != '-1':
-            if choose == '1':
-                balance = user.get_balance(id_num, bal_dict)
-                print('your balance is {}'.format(balance))
-                print('Thank you for using our sevices\n'
-                      'For this action '
-                      'you will be charged for 300$ comission\n'
-                      'Bye Bye!')
-                break
-
-            elif choose == '2':
-                amount = input('How much would you like to deposit?\n')
-                user.set_deposit(id_num, amount, pass_dict, bal_dict)
-                print('now your balance is {}'.format(user.get_balance
-                                                      (id_num, bal_dict)))
-                break
-            elif choose == '3':
-                amount = input('How much would you like to withdraw?\n')
-                user.set_withdraw(id_num, amount, pass_dict, bal_dict)
-                print('now your balance is {}'.format
-                      (user.get_balance(id_num, bal_dict)))
-                break
-            elif choose == '4':
-                new_password = '0'
-                while len(new_password) != 4:
-                    new_password = input('please enter 4 digit password\n')
-                user.set_change_password(new_password, id_num, pass_dict,
-                                         bal_dict)
-                print('password successfully changed')
-                break
-            else:
-                print('invalid input')
-                break
+    if id_num in user.pass_dict.keys():
+        user.id_num = id_num
+        user.password = input('please enter your password\n')
+        user.check_password(id_num, user.password, user.pass_dict)
+
+        balance_choice = '1'
+        deposit_choice = '2'
+        withdraw_choice = '3'
+        change_pass_choice = '4'
+        exit_choice = '-1'
+        user.choice = user.select()
+        if user.choice == balance_choice:
+            balance = user.get_balance(id_num, user.balance_dict)
```

Comment:
I want you to go over all of your functions and look at their variables.
all the point of making a class variable is that you can use it from the functions in the class, here you keep passing those variables to your functions.

Reply:
done